var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -344,16 +361,21 @@ def mpi_dependent_options(self):
                 '--with-cc=%s' % os.environ['CC'],
                 '--with-cxx=%s' % (os.environ['CXX']
                                    if self.compiler.cxx is not None else '0'),
-                '--with-fc=%s' % (os.environ['FC']
-                                  if self.compiler.fc is not None else '0'),
                 '--with-mpi=0'
             ]
+            if '+fortran' in self.spec and self.compiler.fc is not None:
+                compiler_opts.append('--with-fc=%s' % os.environ['FC'])
+            else:
+                compiler_opts.append('--with-fc=0')
         else:
             compiler_opts = [
                 '--with-cc=%s' % self.spec['mpi'].mpicc,
                 '--with-cxx=%s' % self.spec['mpi'].mpicxx,
-                '--with-fc=%s' % self.spec['mpi'].mpifc,
-            ]
+                ]
```

Comment:
This change is perhaps not intended? [the extra space might affect python syntax?]

Reply:
Reverted the location of the closing brace.